peer_heap.js

```diff
@@ -104,13 +104,15 @@ PeerHeap.prototype.choose = function choose(threshold, filter) {
 
         var el = self.array[i];
 
-        if (self.rangehis[i] <= self.maxRangeStart) {
+        if (!filter && self.rangehis[i] <= self.maxRangeStart) {
```

Comment:
I need to think about it more but I'm not sure that this is what we want here. This actually removes any benefit from using the heap and devolves it into a filtered linear scan.